src/gps/GPS.cpp

```diff
@@ -192,6 +198,147 @@ bool GPS::setupGPS()
             delay(250);
         } else if (gnssModel == GNSS_MODEL_UBLOX) {
 
+            uint8_t CK_A = 0, CK_B = 0; // checksum bytes
+
+            // Configure GNSS system to GPS+SBAS+GLONASS (Module may restart after this command)
+            // We need set it because by default it is GPS only, and we want to use GLONASS too
+            // Also we need SBAS for better accuracy and extra features
+            // ToDo: Dynamic configure GNSS systems depending of LoRa region
+            byte _message_GNSS[36] = {
+                0xb5, 0x62, // Sync message for UBX protocol
+                0x06, 0x3e, // Message class and ID (UBX-CFG-GNSS)
+                0x1c, 0x00, // Length of payload (28 bytes)
+                0x00,       // msgVer (0 for this version)
+                0x00,       // numTrkChHw (max number of hardware channels, read only, so it's always 0)
+                0xff,       // numTrkChUse (max number of channels to use, 0xff = max available)
+                0x03,       // numConfigBlocks (number of GNSS systems), most modules support maximum 3 GNSS systems
+                // GNSS config format: gnssId, resTrkCh, maxTrkCh, reserved1, flags
+                0x00, 0x08, 0x10, 0x00, 0x01, 0x00, 0x01, 0x01, // GPS
+                0x01, 0x01, 0x03, 0x00, 0x01, 0x00, 0x01, 0x01, // SBAS
+                0x06, 0x08, 0x0e, 0x00, 0x01, 0x00, 0x01, 0x01, // GLONASS
+                0x00, 0x00                                      // Checksum (to be calculated below)
+            };
+
+            // Reset the checksum values
+            CK_A = 0;
+            CK_B = 0;
+
+            // Calculate the checksum, starting from the CLASS field (which is _message_GNSS[2])
+            for (size_t i = 2; i < sizeof(_message_GNSS) - 2; i++) {
+                CK_A = (CK_A + _message_GNSS[i]) & 0xFF;
+                CK_B = (CK_B + CK_A) & 0xFF;
+            }
+
+            // Place the calculated checksum values in the message
+            _message_GNSS[sizeof(_message_GNSS) - 2] = CK_A;
+            _message_GNSS[sizeof(_message_GNSS) - 1] = CK_B;
+
```

Comment:
My suggestion would be to make a function to calculate the checksum of a buffer and send to the chip. This block is repeated during init

Reply:
Added in 5b75af1